apiserver/application/application_unit_test.go

```diff
@@ -98,16 +88,72 @@ func (s *ApplicationSuite) SetUpTest(c *gc.C) {
 		nil,
 	)
 	c.Assert(err, jc.ErrorIsNil)
-	s.api = api
+	s.api = application.NewAPIv5(api)
+}
+
+func (s *ApplicationSuite) setUpBasicTest() {
+	ch := mockCharm{
+		config: &charm.Config{
+			Options: map[string]charm.Option{
+				"stringOption": {Type: "string"},
+				"intOption":    {Type: "int", Default: int(123)},
+			},
+		},
+		meta: &charm.Meta{
+			Series: []string{"trusty", "xenial", "yakkety"},
+		},
+	}
+	app := mockApplication{
+		units: []mockUnit{{
+			tag: names.NewUnitTag("foo/0"),
+		}, {
+			tag: names.NewUnitTag("foo/1"),
+		}},
+		charmURL: "cs:foo",
+		charm:    &ch,
+	}
+
+	s.applications["foo"] = &app
```

Comment:
The tests were originally setup that way, however it appears that neither s.applications nor s.charms are not needed, removed.  Looking at the other mock structs in the ApplicationSuite, endpoints and relation are not needed either, removed those as well.